lib/util.js

```diff
@@ -616,8 +616,23 @@ exports.log = function() {
  * @param {function} ctor Constructor function which needs to inherit the
  *     prototype.
  * @param {function} superCtor Constructor function to inherit prototype from.
+ * @throws {TypeError} Will error if either constructor is null, or if
+ *     the super constructor lacks a prototype.
  */
 exports.inherits = function(ctor, superCtor) {
+
+  if (ctor === undefined || ctor === null)
```

Comment:
That in turn triggers a jshint error. Wasn't feeling ballsy enough to add a jshint ignore line in my first PR :P